boilerplate/app/screens/LoginScreen.tsx

```diff
@@ -0,0 +1,138 @@
+import { observer } from "mobx-react-lite"
```

Comment:
The LoginScreen has a few features:
- hooks up text-fields to an MST store
- shows KeyboardAvoidingView in action
- shows field validations, delaying them until the first submit click
- shows a hint after a few failed submissions
- focuses fields / submits when keyboard return key is tapped
- shows memoization of components to prevent re-renders when text-field is updated